code/game/objects/items/devices/swapper.dm

```diff
@@ -0,0 +1,108 @@
+/obj/item/swapper
+	name = "quantum spin inverter"
+	desc = "An experimental device that is able to swap the locations of two entities by switching their particles' spin values. Must be linked to another device to function."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "swapper"
+	item_state = "electronic"
+	w_class = WEIGHT_CLASS_SMALL
+	item_flags = NOBLUDGEON
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+
+	var/cooldown = 300
+	var/next_use = 0
+	var/obj/item/swapper/linked_swapper
+
+/obj/item/swapper/Destroy()
+	if(linked_swapper)
+		linked_swapper.linked_swapper = null //*inception music*
+		linked_swapper.update_icon()
+		linked_swapper = null
+	return ..()
+
+/obj/item/swapper/update_icon()
+	if(linked_swapper)
+		icon_state = "swapper-linked"
+	else
+		icon_state = "swapper"
+	..()
+
+/obj/item/swapper/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/swapper))
+		var/obj/item/swapper/other_swapper = I
+		if(other_swapper.linked_swapper)
+			to_chat(user, "<span class='warning'>[other_swapper] is already linked. Break the current link to establish a new one.</span>")
+			return
+		if(linked_swapper)
+			to_chat(user, "<span class='warning'>[src] is already linked. Break the current link to establish a new one.</span>")
+			return
+		to_chat(user, "<span class='notice'>You establish a quantum link between the two devices.</span>")
+		linked_swapper = other_swapper
+		other_swapper.linked_swapper = src
+		update_icon()
+		linked_swapper.update_icon()
+	else
+		return ..()
+
+/obj/item/swapper/attack_self(mob/living/user)
+	if(world.time < next_use)
+		to_chat(user, "<span class='warning'>[src] is still recharging.</span>")
+		return
+	if(!linked_swapper)
+		to_chat(user, "<span class='warning'>[src] is not linked with another swapper.</span>")
+		return
+	if(world.time < linked_swapper.cooldown)
```

Comment:
Guessing you meant `next_use` instead of `cooldown` here

Reply:
That's right, although i found out that balance-wise it's better to have the cooldowns separate anyway. I'll be removing the check altogether.